tsl/src/continuous_aggs/create.c

```diff
@@ -298,6 +334,76 @@ cagg_create_hypertable(int32 hypertable_id, Oid mat_tbloid, const char *matpartc
 				 errmsg("could not create materialization hypertable")));
 }
 
+/*
+ * Add a logical decoding slot for reading invalidations.
+ *
+ * If a logical slot do not exist, it will be created. If it already exists,
+ * nothing will happen.
+ *
+ * This code is mostly copied from create_logical_replication_slot and friends
+ * in slotfunc.c.
+ *
+ * It is important that the slot exists if and only if there are continuous
+ * aggregates that use the WAL to collect invalidations.
+ *
+ * Obviously, the slot is needed to read invalidations for the hypertables and
+ * it need to have the version expected by the timescale extension. If the
+ * versions do not match, the returned rows will not match what is expected.
+ *
+ * Less obvious, the slot need to be removed once there are no more continuous
+ * aggregates that need the slot. If not, the WAL will not be used and will
+ * start to fill up storage.
+ *
+ * There is a bug in PostgreSQL causing it to error out if you have too many
+ * "old" transactions that need to be cleaned up, so running this when you
+ * have a lot of concurrent transactions might cause it to error out before
+ * the slot is created.
+ */
+static void
+cagg_add_logical_decoding_slot(void)
+{
+	CatalogSecurityContext sec_ctx;
+	LogicalDecodingContext *ctx = NULL;
+	const char *slot_name = ts_invalidation_get_slot_name();
+	const char *plugin_name = CONTINUOUS_AGGS_HYPERTABLE_INVALIDATION_PLUGIN_NAME;
+
+	if (SearchNamedReplicationSlot(slot_name, true) != NULL)
+		return;
+
+	Assert(!MyReplicationSlot);
+
+	ts_catalog_database_info_become_owner(ts_catalog_database_info_get(), &sec_ctx);
+
+	TS_DEBUG_LOG("creating replication slot %s", slot_name);
+#if PG17_LT
+	ReplicationSlotCreate(slot_name, true, RS_EPHEMERAL, false);
```

Comment:
At first I thought that using emphemeral slots would be wrong, but then I read the PG code, and it seems the slot is made persistent later with `ReplicationSlotPersists()` called below. I suggest adding a note about this.